src/edu/ucsb/nceas/metacat/dataone/quota/QuotaService.java

```diff
@@ -0,0 +1,259 @@
+     * @throws ServiceFailure
+     * @throws InvalidRequest
+     * @throws InsufficientResources
+     * @throws NotImplemented
+     */
+    public abstract void enforce(String quotaSubject, Subject requestor, String instanceId, SystemMetadata sysmeta, String method) throws ServiceFailure, InvalidRequest, InsufficientResources, NotImplemented, NotFound, UnsupportedEncodingException ;
+
+    /**
+     * Checking if the given pid is last one in this series chain hasn't been archived
+     * @param pid  the pid of the object will be checked
+     * @param sid  the id of the series chain will be checked
+     * @return true if the pid is the last one hasn't been archived; otherwise fals.
+     * @throws SQLException 
+     * @throws InvalidRequest 
+     * @throws ServiceFailure 
+     */
+    protected boolean isLastUnarchivedInChain(String pid, String sid) throws InvalidRequest, ServiceFailure {
+        boolean lastOne = true;
+        if (sid != null && !sid.trim().equals("") && pid != null && !pid.trim().equals("")) {
+            try {
+                List<String> guids = IdentifierManager.getInstance().getAllPidsInChain(sid);
+                for (String guid : guids) {
+                    if (!guid.equals(pid)) {
+                        Identifier identifier = new Identifier();
+                        identifier.setValue(guid);
+                        SystemMetadata sysmeta = HazelcastService.getInstance().getSystemMetadataMap().get(identifier);
+                        if(sysmeta.getArchived() == null || !sysmeta.getArchived()) {
+                            lastOne = false;//found one which is not archived and its guid doesn't equals the pid
+                            logMetacat.debug("QuotaService.isLastUnarchivedInChain - found the guid " + guid + " in the chain with sid " + sid +" hasn't been archived. So the whole chain hasn't been archived.");
+                            break;
+                        }
+                    }
+                }
+            } catch (SQLException e) {
+                throw new ServiceFailure("1104", "QuotaService.isLastUnarchivedInChain - Can't get the pids list in the chain with the sid " + sid + " since " + e.getMessage());
+            }
+            
+        } else {
+            throw new InvalidRequest("1102", "QuotaService.isLastUnarchivedInChain - the pid or sid can't be null or blank for the portal quota.");
+        }
+        return lastOne;
+    }
+    
+    /**
+     * Checking if the given pid is last one in this series chain hasn't been deleted
+     * @param pid  the pid of the object will be checked
+     * @param sid  the id of the series chain will be checked
+     * @return true if the pid is the last one hasn't been deleted; otherwise fals.
+     * @throws SQLException 
+     * @throws InvalidRequest 
+     * @throws ServiceFailure 
+     */
+    protected boolean isLastUnDletedInChain(String pid, String sid) throws InvalidRequest, ServiceFailure {
```

Comment:
Minor typo here and where it is called.
```suggestion
    protected boolean isLastUndeletedInChain(String pid, String sid) throws InvalidRequest, ServiceFailure {
```

Reply:
See the commit b1a267ef091b34770080a7a67ffda970fe6ab2be.